token-swap/program/src/processor.rs

```diff
@@ -672,6 +676,15 @@ impl PrintProgramError for SwapError {
     }
 }
 
+fn to_u128(val: u64) -> Result<u128, SwapError> {
+    val.try_into().map_err(|_| SwapError::CalculationFailure)
```

Comment:
Is it ever possible for this to fail?

Reply:
Probably not, the compiler should figure it out though.

Reply:
id rather not "unwrap" or throw a random exception since the exception makes it hard to reason about all the exit conditions.